ambari-server/src/main/resources/stack-hooks/before-INSTALL/scripts/params.py

```diff
@@ -58,21 +58,21 @@
 #hosts
 hostname = execution_command.get_host_name()
 ambari_server_hostname = execution_command.get_ambari_server_host()
-rm_host = execution_command._execution_command.__get_value("clusterHostInfo/resourcemanager_hosts", [])
-slave_hosts = execution_command._execution_command.__get_value("clusterHostInfo/datanode_hosts", [])
-oozie_servers = execution_command._execution_command.__get_value("clusterHostInfo/oozie_server", [])
-hcat_server_hosts = execution_command._execution_command.__get_value("clusterHostInfo/webhcat_server_hosts", [])
-hive_server_host =  execution_command._execution_command.__get_value("clusterHostInfo/hive_server_hosts", [])
-hbase_master_hosts = execution_command._execution_command.__get_value("clusterHostInfo/hbase_master_hosts", [])
-hs_host = execution_command._execution_command.__get_value("clusterHostInfo/historyserver_hosts", [])
-jtnode_host = execution_command._execution_command.__get_value("clusterHostInfo/jtnode_hosts", [])
-namenode_host = execution_command._execution_command.__get_value("clusterHostInfo/namenode_hosts", [])
-zk_hosts = execution_command._execution_command.__get_value("clusterHostInfo/zookeeper_server_hosts", [])
-ganglia_server_hosts = execution_command._execution_command.__get_value("clusterHostInfo/ganglia_server_hosts", [])
-storm_server_hosts = execution_command._execution_command.__get_value("clusterHostInfo/nimbus_hosts", [])
-falcon_host = execution_command._execution_command.__get_value("clusterHostInfo/falcon_server_hosts", [])
-
-has_sqoop_client = 'sqoop-env' in module_configs
+rm_host = execution_command.get_resourcemanager_hosts()
+slave_hosts = execution_command.get_datanode_hosts()
+oozie_servers = execution_command.get_oozie_server_hosts
+hcat_server_hosts = execution_command.get_webhcat_server_hosts()
+hive_server_host =  execution_command.get_hive_server_hosts()
+hbase_master_hosts = execution_command.get_hbase_master_hosts()
+hs_host = execution_command.get_historyserver_hosts()
+jtnode_host = execution_command.get_jtnode_hosts()
+namenode_host = execution_command.get_namenode_hosts()
+zk_hosts = execution_command.get_zk_server_hosts()
+ganglia_server_hosts = execution_command.get_ganglia_server_hosts()
+storm_server_hosts = execution_command.get_nimbus_hosts()
+falcon_host = execution_command.get_falcon_server_hosts()
+
+has_sqoop_client = module_configs.get_all_properties(module_name, 'sqoop-env') is not None
```

Comment:
get_all_properties() never returns None. It returns an empty map if the config-type does not exist in the command.json